arch/xtensa/core/cache.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/toolchain.h>
+
+#include <zephyr/arch/xtensa/cache.h>
+#include <zephyr/cache.h>
+
+#ifdef CONFIG_DCACHE
+
+void arch_dcache_enable(void)
+{
+	/* not implemented */
+}
+
+int arch_dcache_range(void *addr, size_t size, int op)
```

Comment:
Can we move this to a header and mark it ALWAYS_INLINE?  The previous API had the advantage of inlining cleanly, so e.g. if you just wanted to flush a pointer to memory of known size and alignment (by far the common case) it would reduce to just one instruction per cache line.  Doing it this way is going to have to invoke the loop code every time, which requires doing a bunch of alignment math on the arguments.

Reply:
Yes, please let's at least allow the compiler eliminate all these branches. In fact I'd actually say the `op` is constant in 100% of the cases, isn't it? When would the operation have to be parametrised and determined at run-time? Don't remember seeing any such cases. In all cases I can think of the caller calls the one and only required operation in that case. So I'm actually not very positive about this API as such...